We should be able to modify the default value that triggers that rule

26 is too much

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: f1148f2d-f598-9a79-2036-aa8b302f8bd3
* Version Independent ID: b88d281d-59fd-353b-3548-ac5f9db0c49d
* Content: [CA1502: Avoid excessive complexity (code analysis) - .NET](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1502)
* Content Source: [docs/fundamentals/code-analysis/quality-rules/ca1502.md](https://github.com/dotnet/docs/blob/main/docs/fundamentals/code-analysis/quality-rules/ca1502.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @gewarren
* Microsoft Alias: **gewarren**

That's weird.

When I try to build I have the following errors:
`2>C:\Program Files\dotnet\sdk\6.0.308\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets(144,5): error NETSDK1045: The current .NET SDK does not support targeting .NET 7.0.  Either target .NET 6.0 or lower, or use a version of the .NET SDK that supports .NET 7.0.
2>Done building project "ReleaseNotesUtil.csproj" -- FAILED.
4>C:\Program Files\dotnet\sdk\6.0.308\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(127,5): error NETSDK1141: Unable to resolve the .NET SDK version as specified in the global.json located at C:\git\roslyn-analyzers\roslyn-analyzers\global.json.`

And when I remove the global.json file I have ther following errors:
`1>C:\git\roslyn-analyzers\roslyn-analyzers\Directory.Build.props(13,3): error : C:\Program Files\dotnet\sdk\6.0.308\Sdks\Microsoft.DotNet.Arcade.Sdk\Sdk not found. Check that a recent enough .NET SDK is installed and/or increase the version specified in global.json.
1>C:\git\roslyn-analyzers\roslyn-analyzers\Directory.Build.props(13,31): error MSB4236: The SDK 'Microsoft.DotNet.Arcade.Sdk' specified could not be found.`

I have 3 SDKs installed:
![image](https://user-images.githubusercontent.com/16304182/215503446-8764b0a3-cc32-4f37-9cb5-b1f9f056b925.png)

and the sources look fine:
![image](https://user-images.githubusercontent.com/16304182/215503562-af3f47bd-b2ab-4f4d-af06-d7716e96cd8b.png)

I tried to fix it and got to https://github.com/dotnet/arcade but I'm not authorized to download it through the provided link https://maestro-prod.westus2.cloudapp.azure.com/
![image](https://user-images.githubusercontent.com/16304182/215504208-3f4d0b9d-bfa2-4be4-9c21-b4a62308d8d7.png)

Any help appreciated!

> That's weird.
> 
> When I try to build I have the following errors: `2>C:\Program Files\dotnet\sdk\6.0.308\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets(144,5): error NETSDK1045: The current .NET SDK does not support targeting .NET 7.0. Either target .NET 6.0 or lower, or use a version of the .NET SDK that supports .NET 7.0. 2>Done building project "ReleaseNotesUtil.csproj" -- FAILED. 4>C:\Program Files\dotnet\sdk\6.0.308\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(127,5): error NETSDK1141: Unable to resolve the .NET SDK version as specified in the global.json located at C:\git\roslyn-analyzers\roslyn-analyzers\global.json.`
> 
> And when I remove the global.json file I have ther following errors: `1>C:\git\roslyn-analyzers\roslyn-analyzers\Directory.Build.props(13,3): error : C:\Program Files\dotnet\sdk\6.0.308\Sdks\Microsoft.DotNet.Arcade.Sdk\Sdk not found. Check that a recent enough .NET SDK is installed and/or increase the version specified in global.json. 1>C:\git\roslyn-analyzers\roslyn-analyzers\Directory.Build.props(13,31): error MSB4236: The SDK 'Microsoft.DotNet.Arcade.Sdk' specified could not be found.`
> 
> I have 3 SDKs installed: ![image](https://user-images.githubusercontent.com/16304182/215503446-8764b0a3-cc32-4f37-9cb5-b1f9f056b925.png)
> 
> and the sources look fine: ![image](https://user-images.githubusercontent.com/16304182/215503562-af3f47bd-b2ab-4f4d-af06-d7716e96cd8b.png)
> 
> I tried to fix it and got to https://github.com/dotnet/arcade but I'm not authorized to download it through the provided link https://maestro-prod.westus2.cloudapp.azure.com/ ![image](https://user-images.githubusercontent.com/16304182/215504208-3f4d0b9d-bfa2-4be4-9c21-b4a62308d8d7.png)
> 
> Any help appreciated!
> 
> EDIT: Actually I can't even load Microsoft.CodeAnalysis.Analyzers ![image](https://user-images.githubusercontent.com/16304182/215507994-968018e2-341e-4445-848d-555559a734f4.png)
> 
> ![image](https://user-images.githubusercontent.com/16304182/215508053-e2304ec1-f8a6-49c9-88c8-7f3f7cf1137a.png)
> 
> ![image](https://user-images.githubusercontent.com/16304182/215508163-48f606cd-98a2-4852-b17c-a3c2eaeaf764.png)
> 
> Looks like `MicrosoftCodeAnalysisVersion` is undefined

@Youssef1313 any idea?

--- a/docs/fundamentals/code-analysis/quality-rules/ca1501.md
+++ b/docs/fundamentals/code-analysis/quality-rules/ca1501.md
@@ -22,16 +22,19 @@ dev_langs:
 | **Rule ID**                         | CA1501                                         |
 | **Category**                        | [Maintainability](maintainability-warnings.md) |
 | **Fix is breaking or non-breaking** | Breaking                                       |
+| **Default threshold**               | 5                                              |
 
 ## Cause
 
-A type is more than four levels deep in its inheritance hierarchy.
+A type is five or more levels deep in its inheritance hierarchy.
 
-By default, the rule only excludes types from the `System` namespace, but this is [configurable](#configure-code-to-analyze). You can also configure the threshold at which this rule fires. For more information, [Configure threshold](#configure-threshold).
+By default, the rule excludes types from the `System` namespace. You can [configure](#configure-code-to-analyze) the rule to exclude other types or namespaces as well.
 
 ## Rule description
 
 Deeply nested type hierarchies can be difficult to follow, understand, and maintain. This rule limits analysis to hierarchies in the same module.
+
+You can configure the inheritance tree depth at which this rule fires. For more information, [Configure threshold](#configure-threshold).
 
 ## How to fix violations
 
@@ -87,7 +90,7 @@ You can configure the threshold at which this rule fires.
    CA1501: 8
    ```
 
-   In this example, the rule is configured to fire when a type is more than eight levels deep in its inheritance hierarchy.
+   In this example, the rule is configured to fire when a type is eight or more levels deep in its inheritance hierarchy.
 
 3. In the project file, mark the build action of the configuration file as [**AdditionalFiles**](/visualstudio/ide/build-actions#build-action-values). For example:
 

--- a/docs/fundamentals/code-analysis/quality-rules/ca1501.md
+++ b/docs/fundamentals/code-analysis/quality-rules/ca1501.md
@@ -79,7 +79,7 @@ For more information, see [How to suppress code analysis warnings](../suppress-w
 
 You can configure the threshold at which this rule fires.
 
-1. Create a text file. As an example, you can name it *CodeMetricsConfig.txt*.
+1. Create a text file named *CodeMetricsConfig.txt*.
 
 2. Add the desired threshold to the text file in the following format:
 

--- a/docs/fundamentals/code-analysis/quality-rules/ca1501.md
+++ b/docs/fundamentals/code-analysis/quality-rules/ca1501.md
@@ -28,13 +28,14 @@ dev_langs:
 
 A type is five or more levels deep in its inheritance hierarchy.
 
-By default, the rule excludes types from the `System` namespace. You can [configure](#configure-code-to-analyze) the rule to exclude other types or namespaces as well.
-
 ## Rule description
 
 Deeply nested type hierarchies can be difficult to follow, understand, and maintain. This rule limits analysis to hierarchies in the same module.
 
-You can configure the inheritance tree depth at which this rule fires. For more information, [Configure threshold](#configure-threshold).
+You can configure this rule in the following ways:
+
+- By default, the rule excludes types from the `System` namespace. You can configure the rule to [exclude other types or namespaces](#configure-code-to-analyze) as well.
+- You can [configure the inheritance tree depth](#configure-threshold) at which this rule fires.
 
 ## How to fix violations
 

--- a/docs/fundamentals/code-analysis/quality-rules/ca1501.md
+++ b/docs/fundamentals/code-analysis/quality-rules/ca1501.md
@@ -27,7 +27,7 @@ dev_langs:
 
 A type is more than four levels deep in its inheritance hierarchy.
 
-By default, the rule only excludes types from the `System` namespace, but this is [configurable](#configure-code-to-analyze).
+By default, the rule only excludes types from the `System` namespace, but this is [configurable](#configure-code-to-analyze). You can also configure the threshold at which this rule fires. For more information, [Configure threshold](#configure-threshold).
 
 ## Rule description
 
@@ -39,7 +39,7 @@ To fix a violation of this rule, derive the type from a base type that is less d
 
 ## When to suppress warnings
 
-It is safe to suppress a warning from this rule. However, the code might be more difficult to maintain. Depending on the visibility of base types, resolving violations of this rule might create breaking changes. For example, removing public base types is a breaking change.
+It's safe to suppress a warning from this rule. However, the code might be more difficult to maintain. Depending on the visibility of base types, resolving violations of this rule might create breaking changes. For example, removing public base types is a breaking change.
 
 > [!NOTE]
 > You might see false positive warnings from this rule if all of the following apply:
@@ -74,6 +74,28 @@ dotnet_analyzer_diagnostic.category-Maintainability.severity = none
 ```
 
 For more information, see [How to suppress code analysis warnings](../suppress-warnings.md).
+
+## Configure threshold
+
+You can configure the threshold at which this rule fires.
+
+1. Create a text file. As an example, you can name it *CodeMetricsConfig.txt*.
+
+2. Add the desired threshold to the text file in the following format:
+
+   ```txt
+   CA1501: 8
+   ```
+
+   In this example, the rule is configured to fire when a type is more than eight levels deep in its inheritance hierarchy.
+
+3. In the project file, mark the build action of the configuration file as [**AdditionalFiles**](/visualstudio/ide/build-actions#build-action-values). For example:
+
+   ```xml
+   <ItemGroup>
+     <AdditionalFiles Include="CodeMetricsConfig.txt" />
+   </ItemGroup>
+   ```
 
 ## Configure code to analyze
 

--- a/docs/fundamentals/code-analysis/quality-rules/ca1502.md
+++ b/docs/fundamentals/code-analysis/quality-rules/ca1502.md
@@ -85,7 +85,7 @@ For more information, see [How to suppress code analysis warnings](../suppress-w
 
 You can configure the threshold at which this rule fires.
 
-1. Create a text file. As an example, you can name it *CodeMetricsConfig.txt*.
+1. Create a text file named *CodeMetricsConfig.txt*.
 
 2. Add the desired threshold to the text file in the following format:
 

--- a/docs/fundamentals/code-analysis/quality-rules/ca1502.md
+++ b/docs/fundamentals/code-analysis/quality-rules/ca1502.md
@@ -22,6 +22,7 @@ dev_langs:
 | **Rule ID**                         | CA1502                                         |
 | **Category**                        | [Maintainability](maintainability-warnings.md) |
 | **Fix is breaking or non-breaking** | Non-breaking                                   |
+| **Default threshold**               | 25                                             |
 
 ## Cause
 
@@ -35,7 +36,7 @@ cyclomatic complexity = the number of edges - the number of nodes + 1
 
 A *node* represents a logic branch point and an *edge* represents a line between nodes.
 
-The rule reports a violation when the cyclomatic complexity is more than 25. However, you can [configure the threshold](#configure-threshold).
+The rule reports a violation when the cyclomatic complexity of a method is more than 25. However, you can [configure the threshold](#configure-threshold) and also specify other kinds of symbols that the rule should analyze.
 
 You can learn more about code metrics at [Measure complexity of managed code](/visualstudio/code-quality/code-metrics-values).
 
@@ -83,7 +84,15 @@ For more information, see [How to suppress code analysis warnings](../suppress-w
 
 ## Configure threshold
 
-You can configure the threshold at which this rule fires.
+You can configure the threshold at which this rule fires and the kinds of symbols to analyze. The allowable symbol kinds are:
+
+- `Assembly`
+- `Namespace`
+- `Type`
+- `Method`
+- `Field`
+- `Event`
+- `Property`
 
 1. Create a text file named *CodeMetricsConfig.txt*.
 
@@ -94,6 +103,12 @@ You can configure the threshold at which this rule fires.
    ```
 
    In this example, the rule is configured to fire when a method's cyclomatic complexity is greater than 10.
+
+   ```txt
+   CA1502(Type): 4
+   ```
+
+   In this example, the rule is configured to fire when a type's cyclomatic complexity is greater than 4. With this configuration file, the rule would continue to report methods with a cyclomatic complexity greater than the default (25).
 
 3. In the project file, mark the build action of the configuration file as [**AdditionalFiles**](/visualstudio/ide/build-actions#build-action-values). For example:
 

--- a/docs/fundamentals/code-analysis/quality-rules/ca1502.md
+++ b/docs/fundamentals/code-analysis/quality-rules/ca1502.md
@@ -35,7 +35,7 @@ cyclomatic complexity = the number of edges - the number of nodes + 1
 
 A *node* represents a logic branch point and an *edge* represents a line between nodes.
 
-The rule reports a violation when the cyclomatic complexity is more than 25.
+The rule reports a violation when the cyclomatic complexity is more than 25. However, you can [configure the threshold](#configure-threshold).
 
 You can learn more about code metrics at [Measure complexity of managed code](/visualstudio/code-quality/code-metrics-values).
 
@@ -81,13 +81,34 @@ dotnet_analyzer_diagnostic.category-Maintainability.severity = none
 
 For more information, see [How to suppress code analysis warnings](../suppress-warnings.md).
 
+## Configure threshold
+
+You can configure the threshold at which this rule fires.
+
+1. Create a text file. As an example, you can name it *CodeMetricsConfig.txt*.
+
+2. Add the desired threshold to the text file in the following format:
+
+   ```txt
+   CA1502: 10
+   ```
+
+   In this example, the rule is configured to fire when a method's cyclomatic complexity is greater than 10.
+
+3. In the project file, mark the build action of the configuration file as [**AdditionalFiles**](/visualstudio/ide/build-actions#build-action-values). For example:
+
+   ```xml
+   <ItemGroup>
+     <AdditionalFiles Include="CodeMetricsConfig.txt" />
+   </ItemGroup>
+   ```
+
 ## How cyclomatic complexity is calculated
 
 The cyclomatic complexity is calculated by adding 1 to the following:
 
-- Number of branches (such as `if`, `while`, and `do`)
-
-- Number of `case` statements in a `switch`
+- The number of branches (such as `if`, `while`, and `do`).
+- The number of `case` statements in a `switch`.
 
 ## Examples
 

--- a/docs/fundamentals/code-analysis/quality-rules/ca1505.md
+++ b/docs/fundamentals/code-analysis/quality-rules/ca1505.md
@@ -72,6 +72,28 @@ dotnet_analyzer_diagnostic.category-Maintainability.severity = none
 
 For more information, see [How to suppress code analysis warnings](../suppress-warnings.md).
 
+## Configure threshold
+
+You can configure the threshold at which this rule fires.
+
+1. Create a text file. As an example, you can name it *CodeMetricsConfig.txt*.
+
+2. Add the desired threshold to the text file in the following format:
+
+   ```txt
+   CA1505: 10
+   ```
+
+   In this example, the rule is configured to fire when a method's maintainability index is less than 10.
+
+3. In the project file, mark the build action of the configuration file as [**AdditionalFiles**](/visualstudio/ide/build-actions#build-action-values). For example:
+
+   ```xml
+   <ItemGroup>
+     <AdditionalFiles Include="CodeMetricsConfig.txt" />
+   </ItemGroup>
+   ```
+
 ## See also
 
 - [Maintainability rules](maintainability-warnings.md)

--- a/docs/fundamentals/code-analysis/quality-rules/ca1505.md
+++ b/docs/fundamentals/code-analysis/quality-rules/ca1505.md
@@ -76,7 +76,7 @@ For more information, see [How to suppress code analysis warnings](../suppress-w
 
 You can configure the threshold at which this rule fires.
 
-1. Create a text file. As an example, you can name it *CodeMetricsConfig.txt*.
+1. Create a text file named *CodeMetricsConfig.txt*.
 
 2. Add the desired threshold to the text file in the following format:
 

--- a/docs/fundamentals/code-analysis/quality-rules/ca1505.md
+++ b/docs/fundamentals/code-analysis/quality-rules/ca1505.md
@@ -19,14 +19,17 @@ ms.author: gewarren
 | **Rule ID**                         | CA1505                                         |
 | **Category**                        | [Maintainability](maintainability-warnings.md) |
 | **Fix is breaking or non-breaking** | Non-breaking                                   |
+| **Default threshold**               | 10                                             |
 
 ## Cause
 
-A type or method has a low maintainability index value.
+A type, method, field, property, or event has a low maintainability index value.
 
 ## Rule description
 
-The maintainability index is calculated by using the following metrics: lines of code, program volume, and cyclomatic complexity. Program volume is a measure of the difficulty of understanding of a type or method that's based on the number of operators and operands in the code. Cyclomatic complexity is a measure of the structural complexity of the type or method. You can learn more about code metrics at [Measure complexity and maintainability of managed code](/visualstudio/code-quality/code-metrics-values).
+The rule reports a violation when the maintainability index of a type, method, field, property, or event is less than 10. However, you can [configure the threshold](#configure-threshold).
+
+The maintainability index is calculated by using the following metrics: lines of code, program volume, and cyclomatic complexity. (*Program volume* is a measure of the difficulty of understanding of a type or method that's based on the number of operators and operands in the code. *Cyclomatic complexity* is a measure of the structural complexity of the type or method. You can learn more about code metrics at [Measure complexity and maintainability of managed code](/visualstudio/code-quality/code-metrics-values).
 
 A low maintainability index indicates that a type or method is probably difficult to maintain and would be a good candidate to redesign.
 
@@ -74,17 +77,32 @@ For more information, see [How to suppress code analysis warnings](../suppress-w
 
 ## Configure threshold
 
-You can configure the threshold at which this rule fires.
+You can configure the threshold at which this rule fires and the kinds of symbols to analyze. The allowable symbol kinds are:
+
+- `Assembly`
+- `Namespace`
+- `Type`
+- `Method`
+- `Field`
+- `Event`
+- `Property`
 
 1. Create a text file named *CodeMetricsConfig.txt*.
 
 2. Add the desired threshold to the text file in the following format:
 
    ```txt
-   CA1505: 10
+   CA1505: 20
    ```
 
-   In this example, the rule is configured to fire when a method's maintainability index is less than 10.
+   In this example, the rule is configured to fire when the maintainability index of a type, method, field, property, or event is less than 20.
+
+   ```txt
+   CA1505(Method): 5
+   CA1505(Type): 15
+   ```
+
+   In this example, the rule is configured to fire when the maintainability index of a method is less than 5 or the maintainability index of a type is less than 15. With this configuration file, the rule will continue to flag fields, properties, and events whose maintainability index is less than the default threshold (10).
 
 3. In the project file, mark the build action of the configuration file as [**AdditionalFiles**](/visualstudio/ide/build-actions#build-action-values). For example:
 

--- a/docs/fundamentals/code-analysis/quality-rules/ca1506.md
+++ b/docs/fundamentals/code-analysis/quality-rules/ca1506.md
@@ -23,7 +23,7 @@ A type or method is coupled with many other types. Compiler-generated types are 
 
 ## Rule description
 
-This rule measures class coupling by counting the number of unique type references that a type or method contains. The default coupling threshold is 95 for types and 40 for methods.
+This rule measures class coupling by counting the number of unique type references that a type or method contains. The default coupling threshold is 95 for types and 40 for methods, and the threshold [is configurable](#configure-threshold).
 
 Types and methods that have a high degree of class coupling can be difficult to maintain. It's a good practice to have types and methods that exhibit low coupling and high cohesion.
 
@@ -68,6 +68,28 @@ dotnet_analyzer_diagnostic.category-Maintainability.severity = none
 ```
 
 For more information, see [How to suppress code analysis warnings](../suppress-warnings.md).
+
+## Configure threshold
+
+You can configure the threshold at which this rule fires.
+
+1. Create a text file. As an example, you can name it *CodeMetricsConfig.txt*.
+
+2. Add the desired threshold to the text file in the following format:
+
+   ```txt
+   CA1506: 120
+   ```
+
+   In this example, the rule is configured to fire when a type has more than 120 unique type references.
+
+3. In the project file, mark the build action of the configuration file as [**AdditionalFiles**](/visualstudio/ide/build-actions#build-action-values). For example:
+
+   ```xml
+   <ItemGroup>
+     <AdditionalFiles Include="CodeMetricsConfig.txt" />
+   </ItemGroup>
+   ```
 
 ## See also
 

--- a/docs/fundamentals/code-analysis/quality-rules/ca1506.md
+++ b/docs/fundamentals/code-analysis/quality-rules/ca1506.md
@@ -73,7 +73,7 @@ For more information, see [How to suppress code analysis warnings](../suppress-w
 
 You can configure the threshold at which this rule fires.
 
-1. Create a text file. As an example, you can name it *CodeMetricsConfig.txt*.
+1. Create a text file named *CodeMetricsConfig.txt*.
 
 2. Add the desired threshold to the text file in the following format:
 

--- a/docs/fundamentals/code-analysis/quality-rules/ca1506.md
+++ b/docs/fundamentals/code-analysis/quality-rules/ca1506.md
@@ -16,16 +16,17 @@ helpviewer_keywords:
 | **Rule ID**                         | CA1506                                         |
 | **Category**                        | [Maintainability](maintainability-warnings.md) |
 | **Fix is breaking or non-breaking** | Breaking                                       |
+| **Default threshold**               | Types: 95&nbsp;&nbsp;Other symbols: 40         |
 
 ## Cause
 
-A type or method is coupled with many other types. Compiler-generated types are excluded from this metric.
+A type, method, field, property, or event is coupled with many other types. Compiler-generated types are excluded from this metric.
 
 ## Rule description
 
-This rule measures class coupling by counting the number of unique type references that a type or method contains. The default coupling threshold is 95 for types and 40 for methods, and the threshold [is configurable](#configure-threshold).
+This rule measures class coupling by counting the number of unique type references that a type, method, field, property, or event contains. The default coupling threshold is 95 for types and 40 for other symbol kinds, and the [thresholds are configurable](#configure-threshold).
 
-Types and methods that have a high degree of class coupling can be difficult to maintain. It's a good practice to have types and methods that exhibit low coupling and high cohesion.
+Types, methods, and other symbols that have a high degree of class coupling can be difficult to maintain. It's a good practice to have types, methods, and other symbols that exhibit low coupling and high cohesion.
 
 ## How to fix violations
 
@@ -71,17 +72,26 @@ For more information, see [How to suppress code analysis warnings](../suppress-w
 
 ## Configure threshold
 
-You can configure the threshold at which this rule fires.
+You can configure the threshold at which this rule fires and the kinds of symbols to analyze. The allowable symbol kinds are:
+
+- `Assembly`
+- `Namespace`
+- `Type`
+- `Method`
+- `Field`
+- `Event`
+- `Property`
 
 1. Create a text file named *CodeMetricsConfig.txt*.
 
 2. Add the desired threshold to the text file in the following format:
 
    ```txt
-   CA1506: 120
+   CA1506(Method): 60
+   CA1506(Type): 120
    ```
 
-   In this example, the rule is configured to fire when a type has more than 120 unique type references.
+   In this example, the rule is configured to fire when a method has more than 60 unique type references or a type has more than 120 unique type references. With this configuration file, the rule will continue to flag fields, properties, and events whose class coupling is greater than the default threshold (40).
 
 3. In the project file, mark the build action of the configuration file as [**AdditionalFiles**](/visualstudio/ide/build-actions#build-action-values). For example:
 

--- a/docs/fundamentals/code-analysis/quality-rules/ca1509.md
+++ b/docs/fundamentals/code-analysis/quality-rules/ca1509.md
@@ -1,7 +1,7 @@
 ---
 title: "CA1509: Invalid entry in code metrics configuration file (code analysis)"
 description: "Learn about code analysis rule CA1509: Invalid entry in code metrics configuration file"
-ms.date: 04/28/2020
+ms.date: 02/13/2023
 ms.topic: reference
 f1_keywords:
 - CA1509
@@ -14,19 +14,19 @@ ms.author: mavasani
 ---
 # CA1509: Invalid entry in code metrics configuration file
 
-| | Value |
-|-|-|
-| **Rule ID** |CA1509|
-| **Category** |[Maintainability](maintainability-warnings.md)|
-| **Fix is breaking or non-breaking** |Non-Breaking|
+|                                     | Value                                          |
+| ----------------------------------- | ---------------------------------------------- |
+| **Rule ID**                         | CA1509                                         |
+| **Category**                        | [Maintainability](maintainability-warnings.md) |
+| **Fix is breaking or non-breaking** | Non-Breaking                                   |
 
 ## Cause
 
-Code metrics rules, such as [CA1501](ca1501.md), [CA1502](ca1502.md), [CA1505](ca1505.md) and [CA1506](ca1506.md), supplied a configuration file named `CodeMetricsConfig.txt` that has an invalid entry.
+A configuration file named *CodeMetricsConfig.txt* has an invalid entry.
 
 ## Rule description
 
-.NET code-quality analyzers implementation of [code metrics](/visualstudio/code-quality/code-metrics-values) analysis rules allow end users to supply an [additional file](https://github.com/dotnet/roslyn/blob/main/docs/analyzers/Using%20Additional%20Files.md) named `CodeMetricsConfig.txt`. This file contains entries to configure code metric thresholds for analysis. Following rules are configurable in this file:
+The analysis rules for [code metrics](/visualstudio/code-quality/code-metrics-values) allow you to supply an [additional file](https://github.com/dotnet/roslyn/blob/main/docs/analyzers/Using%20Additional%20Files.md) in your project named *CodeMetricsConfig.txt*. This file contains entries to configure code-metric thresholds for analysis. The following rules are configurable in this file:
 
 - [CA1501: Avoid excessive inheritance](ca1501.md)
 - [CA1502: Avoid excessive complexity](ca1502.md)
@@ -39,10 +39,10 @@ This configuration file expects each entry to be in following format:
 'RuleId'(Optional 'SymbolKind'): 'Threshold'
 ```
 
-- Valid values for 'RuleId' are `CA1501`, `CA1502`, `CA1505`, and `CA1506`.
-- Valid values for optional 'SymbolKind' are `Assembly`, `Namespace`, `Type`, `Method`, `Field`, `Event`, and `Property`.
-- Valid values for 'Threshold' are non-negative integers.
-- Lines starting with '#' are treated as comment lines
+- Valid values for `RuleId` are `CA1501`, `CA1502`, `CA1505`, and `CA1506`.
+- Valid values for the optional `SymbolKind` are `Assembly`, `Namespace`, `Type`, `Method`, `Field`, `Event`, and `Property`.
+- Valid values for `Threshold` are non-negative integers.
+- Lines starting with '#' are treated as comment lines.
 
 For example, the following is a valid configuration file:
 

